src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -291,7 +291,7 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
         {
             try
             {
-                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: project.Documents.Any(d => d.IsOpen())))
+                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: false))
```

Comment:
```
private bool CheckOption(Workspace workspace, string language, bool documentOpened)
{
    var optionService = workspace.Services.GetService<IOptionService>();
    if (optionService == null || optionService.GetOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, language))
    {
        return true;
    }

    if (documentOpened)
    {
        return true;
    }

    return false;
}
```

What I meant was when closed file diagnostic is off, first if statement will always be false and if documentOpened flag is always false, the method will always return false. Current code instead makes us to run project analysis if the given project has at least one open document, which seems correct to me.